compiler/src/iree/compiler/Codegen/Common/TransformDialectInterpreterPass.cpp

```diff
@@ -278,12 +323,50 @@ llvm::raw_ostream &TransformDialectInterpreterPass::printIreeOptReproCall(
 llvm::raw_ostream &TransformDialectInterpreterPass::printModuleForRepro(
     llvm::raw_ostream &os, Operation *root, Operation *transformContainer) {
   root->print(os);
-  if (!root->isAncestor(transformContainer)) {
+  if (transformContainer && !root->isAncestor(transformContainer)) {
     transformContainer->print(os);
   }
   return os;
 }
 
+/// To make the flow compose with "iree-compile -compile-mode=hal-executable",
+/// we must have a single HAL::ExecutablOp in the module **and** the
+/// func::FuncOp that dispatches into it.
+/// The uniqueness also makes sense for us because only one such op is tagged by
+/// the transform dialect repro.
+// TODO: revisit this filtering in the future.
+// TODO: there must be a better way than rolling our own manually and taking the
```

Comment:
@benvanik any words of wisdom to share here ?

Reply:
Adding a bunch of TODOs around something that is clearly wrong doesn't magically allow it all to get merged to main :/ This is very unfortunate.